invisible_cities/io/mcinfo_io.py

```diff
@@ -168,6 +173,30 @@ def __call__(self, mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
         self.generator_table.flush()
 
 
+def copy_mc_info(h5in : tb.File, writer : Type[mc_info_writer], which_events : List[int]=None):
+    """Copy from the input file to the output file the MC info of certain events.
+
+    Parameters
+    ----------
+    h5in  : tb.File
+        Input h5 file.
+    writer : instance of class mcinfo_io.mc_info_writer
+        MC info writer to h5 file.
+    which_events : None or list of ints
+        List of IDs (i.e. event numbers) that identify the events to be copied
+        to the output file. If None, all events in the input file are copied.
+    """
+
+    try:
+        if which_events is None:
+            which_events = h5in.root.MC.extents.cols.evt_number[:]
+        mcinfo = tbl.get_mc_info(h5in)
+        for n in which_events:
+            writer(mctables=mcinfo, evt_number=n)
+    except NoParticleInfoInFile:
+        pass
```

Comment:
Again, I don't think this should pass silently. I think we want it to crash whenever there is a problem in the input data. That's why we have `if run_number <= 0 ` to deal with that case in the cities.